## Условие 1: 1 задача


### Скачать данные kc_house_data (см. вложения)
Считать д`анные с помощью pandas

In [185]:
import pandas as pd

In [186]:
df = pd.read_csv('kc_house_data.csv')

### Вывести на экран первые 5 строк

In [187]:
df.head()

id             date     price  bedrooms  bathrooms  sqft_living  \
0  7129300520  20141013T000000  221900.0         3       1.00         1180   
1  6414100192  20141209T000000  538000.0         3       2.25         2570   
2  5631500400  20150225T000000  180000.0         2       1.00          770   
3  2487200875  20141209T000000  604000.0         4       3.00         1960   
4  1954400510  20150218T000000  510000.0         3       2.00         1680   

   sqft_lot  floors  waterfront  view  ...  grade  sqft_above  sqft_basement  \
0      5650     1.0           0     0  ...      7        1180              0   
1      7242     2.0           0     0  ...      7        2170            400   
2     10000     1.0           0     0  ...      6         770              0   
3      5000     1.0           0     0  ...      7        1050            910   
4      8080     1.0           0     0  ...      8        1680              0   

   yr_built  yr_renovated  zipcode      lat     long  sqft_living15  \
0      1955             0    98178  47.5112 -122.257           1340   
1      1951          1991    98125  47.7210 -122.319           1690   
2      1933             0    98028  47.7379 -122.233           2720   
3      1965             0    98136  47.5208 -122.393           1360   
4      1987             0    98074  47.6168 -122.045           1800   

   sqft_lot15  
0        5650  
1        7639  
2        8062  
3        5000  
4        7503  

[5 rows x 21 columns]

In [188]:
df.sample()

id             date     price  bedrooms  bathrooms  \
15073  8825900020  20140811T000000  925000.0         5        3.0   

       sqft_living  sqft_lot  floors  waterfront  view  ...  grade  \
15073         2710      4200     2.0           0     0  ...      7   

       sqft_above  sqft_basement  yr_built  yr_renovated  zipcode      lat  \
15073        1890            820      1919          2014    98115  47.6754   

          long  sqft_living15  sqft_lot15  
15073 -122.307           2150        4200  

[1 rows x 21 columns]

### Посмотреть на описание признаков и на их содержание

In [189]:
df.describe()

id         price      bedrooms     bathrooms   sqft_living  \
count  2.161300e+04  2.161300e+04  21613.000000  21613.000000  21613.000000   
mean   4.580302e+09  5.400881e+05      3.370842      2.114757   2079.899736   
std    2.876566e+09  3.671272e+05      0.930062      0.770163    918.440897   
min    1.000102e+06  7.500000e+04      0.000000      0.000000    290.000000   
25%    2.123049e+09  3.219500e+05      3.000000      1.750000   1427.000000   
50%    3.904930e+09  4.500000e+05      3.000000      2.250000   1910.000000   
75%    7.308900e+09  6.450000e+05      4.000000      2.500000   2550.000000   
max    9.900000e+09  7.700000e+06     33.000000      8.000000  13540.000000   

           sqft_lot        floors    waterfront          view     condition  \
count  2.161300e+04  21613.000000  21613.000000  21613.000000  21613.000000   
mean   1.510697e+04      1.494309      0.007542      0.234303      3.409430   
std    4.142051e+04      0.539989      0.086517      0.766318      0.650743   
min    5.200000e+02      1.000000      0.000000      0.000000      1.000000   
25%    5.040000e+03      1.000000      0.000000      0.000000      3.000000   
50%    7.618000e+03      1.500000      0.000000      0.000000      3.000000   
75%    1.068800e+04      2.000000      0.000000      0.000000      4.000000   
max    1.651359e+06      3.500000      1.000000      4.000000      5.000000   

              grade    sqft_above  sqft_basement      yr_built  yr_renovated  \
count  21613.000000  21613.000000   21613.000000  21613.000000  21613.000000   
mean       7.656873   1788.390691     291.509045   1971.005136     84.402258   
std        1.175459    828.090978     442.575043     29.373411    401.679240   
min        1.000000    290.000000       0.000000   1900.000000      0.000000   
25%        7.000000   1190.000000       0.000000   1951.000000      0.000000   
50%        7.000000   1560.000000       0.000000   1975.000000      0.000000   
75%        8.000000   2210.000000     560.000000   1997.000000      0.000000   
max       13.000000   9410.000000    4820.000000   2015.000000   2015.000000   

            zipcode           lat          long  sqft_living15     sqft_lot15  
count  21613.000000  21613.000000  21613.000000   21613.000000   21613.000000  
mean   98077.939805     47.560053   -122.213896    1986.552492   12768.455652  
std       53.505026      0.138564      0.140828     685.391304   27304.179631  
min    98001.000000     47.155900   -122.519000     399.000000     651.000000  
25%    98033.000000     47.471000   -122.328000    1490.000000    5100.000000  
50%    98065.000000     47.571800   -122.230000    1840.000000    7620.000000  
75%    98118.000000     47.678000   -122.125000    2360.000000   10083.000000  
max    98199.000000     47.777600   -121.315000    6210.000000  871200.000000

## Условие 2: 2 задача

### Проведите первичный анализ данных


### Изучите типы данных

In [190]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 21613 entries, 0 to 21612
Data columns (total 21 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   id             21613 non-null  int64  
 1   date           21613 non-null  object 
 2   price          21613 non-null  float64
 3   bedrooms       21613 non-null  int64  
 4   bathrooms      21613 non-null  float64
 5   sqft_living    21613 non-null  int64  
 6   sqft_lot       21613 non-null  int64  
 7   floors         21613 non-null  float64
 8   waterfront     21613 non-null  int64  
 9   view           21613 non-null  int64  
 10  condition      21613 non-null  int64  
 11  grade          21613 non-null  int64  
 12  sqft_above     21613 non-null  int64  
 13  sqft_basement  21613 non-null  int64  
 14  yr_built       21613 non-null  int64  
 15  yr_renovated   21613 non-null  int64  
 16  zipcode        21613 non-null  int64  
 17  lat            21613 non-null  float64
 18  long  

### Найдите количество пропущенных ячеек в данных


In [191]:
df.isnull().sum()

id               0
date             0
price            0
bedrooms         0
bathrooms        0
sqft_living      0
sqft_lot         0
floors           0
waterfront       0
view             0
condition        0
grade            0
sqft_above       0
sqft_basement    0
yr_built         0
yr_renovated     0
zipcode          0
lat              0
long             0
sqft_living15    0
sqft_lot15       0
dtype: int64

Пустых ячеек нет 


### Посчитайте основные статистики по всем признакам и поизучайте их


In [192]:
df.describe().round(2)

id       price  bedrooms  bathrooms  sqft_living    sqft_lot  \
count  2.161300e+04    21613.00  21613.00   21613.00     21613.00    21613.00   
mean   4.580302e+09   540088.14      3.37       2.11      2079.90    15106.97   
std    2.876566e+09   367127.20      0.93       0.77       918.44    41420.51   
min    1.000102e+06    75000.00      0.00       0.00       290.00      520.00   
25%    2.123049e+09   321950.00      3.00       1.75      1427.00     5040.00   
50%    3.904930e+09   450000.00      3.00       2.25      1910.00     7618.00   
75%    7.308900e+09   645000.00      4.00       2.50      2550.00    10688.00   
max    9.900000e+09  7700000.00     33.00       8.00     13540.00  1651359.00   

         floors  waterfront      view  condition     grade  sqft_above  \
count  21613.00    21613.00  21613.00   21613.00  21613.00    21613.00   
mean       1.49        0.01      0.23       3.41      7.66     1788.39   
std        0.54        0.09      0.77       0.65      1.18      828.09   
min        1.00        0.00      0.00       1.00      1.00      290.00   
25%        1.00        0.00      0.00       3.00      7.00     1190.00   
50%        1.50        0.00      0.00       3.00      7.00     1560.00   
75%        2.00        0.00      0.00       4.00      8.00     2210.00   
max        3.50        1.00      4.00       5.00     13.00     9410.00   

       sqft_basement  yr_built  yr_renovated   zipcode       lat      long  \
count       21613.00  21613.00      21613.00  21613.00  21613.00  21613.00   
mean          291.51   1971.01         84.40  98077.94     47.56   -122.21   
std           442.58     29.37        401.68     53.51      0.14      0.14   
min             0.00   1900.00          0.00  98001.00     47.16   -122.52   
25%             0.00   1951.00          0.00  98033.00     47.47   -122.33   
50%             0.00   1975.00          0.00  98065.00     47.57   -122.23   
75%           560.00   1997.00          0.00  98118.00     47.68   -122.12   
max          4820.00   2015.00       2015.00  98199.00     47.78   -121.32   

       sqft_living15  sqft_lot15  
count       21613.00    21613.00  
mean         1986.55    12768.46  
std           685.39    27304.18  
min           399.00      651.00  
25%          1490.00     5100.00  
50%          1840.00     7620.00  
75%          2360.00    10083.00  
max          6210.00   871200.00

## Условие 3: 3 задача

### 3.1 В каком диапазоне изменяются стоимости недвижимости?

In [193]:
df_price_min=df['price'].min()
df_price_max=df['price'].max()
print(df_price_min, df_price_max)

75000.0 7700000.0


От 75 000 до 7 700 000 единиц цены

### 3.2 Какую долю в среднем занимают жилая площадь от всей площади по всем домам?

In [194]:
df['living_sqrt_percent'] = df["sqft_living"] / df["sqft_lot"]
df['living_sqrt_percent'].mean()

0.32374472542047195

В среднем жилая площадь занимает 32,3 %

### 3.3 Как много домов с разными этажами в данных?

In [195]:
df['floors'].nunique()

6

In [196]:
df['floors'].value_counts()

1.0    10680
2.0     8241
1.5     1910
3.0      613
2.5      161
3.5        8
Name: floors, dtype: int64

6 видов домов с разными этажами

### 3.4 Насколько хорошие состояния у домов в данных

In [197]:
print(df['condition'].min())
print(df['condition'].max())
print(df['condition'].mean())
print(df['condition'].median())

1
5
3.4094295100171195
3.0


Дома оцениваются по пятибальной системе, в среднем дома имеют оценку 3,4. Медианное значение -3 балла

### 3.5 Найдите года, когда построили первый дом, когда построили последний дом в данных?

In [198]:
print(df['yr_built'].min())
print(df['yr_built'].max())

1900
2015


В данных дома от 1900 до 2015 года постройки

## Условие 4: 4 задача

### 4.1 Сколько в среднем стоят дома, у которых 2 спальни?

In [199]:
mean_price_bedrooms=df[df['bedrooms']== 2 ]['price'].mean()
mean_price_bedrooms

401372.681884058

Дома с двумя спальнями стоят 401372,68

### 4.2 Какая в среднем общая площадь домов, у которых стоимость больше 600 000?

In [200]:
mean_sqrt_price=df[df['price'] > 600000 ]['sqft_lot'].mean()
mean_sqrt_price

20442.524776214832

Дома с ценой больше 600000 имеют среднюю площадь 20442,52

### 4.3 Как много домов коснулся ремонт?

In [201]:
renov= df[~(df['yr_renovated'] == 0)]['yr_renovated'].count()
renov

914

914 домов имеют ремонт

### 4.4 Насколько в среднем стоимость домов с оценкой grade домов выше 10 отличается от стоимости домов с оценкой grade меньше 4?

In [202]:
grade10= df[(df['grade'] > 10)]['price'].mean()
grade4= df[(df['grade'] < 4)]['price'].mean()
grade10/grade4

8.846561884215443

В среднем цена различается в 8,84 раза

## Условие 5: 5 задача

### 5.1 Выберите дом клиенту
Клиент хочет дом с видом на набережную, как минимум с тремя ванными и с подвалом. Сколько вариантов есть у клиента?

In [203]:
df[ (df["bathrooms"] > 2) &
    (df["waterfront"] == 1) &
    (df["sqft_basement"] > 0)
]

id             date      price  bedrooms  bathrooms  \
49      822039084  20150311T000000  1350000.0         3       2.50   
246    2025069065  20140929T000000  2400000.0         4       2.50   
300    3225069065  20140624T000000  3075000.0         4       5.00   
457    2122039094  20141126T000000   705000.0         3       3.00   
656    3760500116  20141120T000000  3070000.0         3       2.50   
...           ...              ...        ...       ...        ...   
18914  9253900408  20150408T000000  1400000.0         3       2.75   
20325   518500480  20140811T000000  3000000.0         3       3.50   
20767  8043700105  20150417T000000  2300000.0         4       4.00   
21201   518500460  20141008T000000  2230000.0         3       3.50   
21576  9253900271  20150107T000000  3567000.0         5       4.50   

       sqft_living  sqft_lot  floors  waterfront  view  ...  sqft_above  \
49            2753     65005     1.0           1     2  ...        2165   
246           3650      8354     1.0           1     4  ...        1830   
300           4550     18641     1.0           1     4  ...        2600   
457           1970     20978     2.0           1     3  ...        1770   
656           3930     55867     1.0           1     4  ...        2330   
...            ...       ...     ...         ...   ...  ...         ...   
18914         3130     19530     1.0           1     4  ...        1690   
20325         4410     10756     2.0           1     4  ...        3430   
20767         4360      8175     2.5           1     4  ...        3940   
21201         3760      5634     2.0           1     4  ...        2830   
21576         4850     10584     2.0           1     4  ...        3540   

       sqft_basement  yr_built  yr_renovated  zipcode      lat     long  \
49               588      1953             0    98070  47.4041 -122.451   
246             1820      2000             0    98074  47.6338 -122.072   
300             1950      2002             0    98074  47.6053 -122.077   
457              200      1980             0    98070  47.3844 -122.438   
656             1600      1957             0    98034  47.7022 -122.224   
...              ...       ...           ...      ...      ...      ...   
18914           1440      1947          1984    98008  47.5895 -122.111   
20325            980      2014             0    98056  47.5283 -122.205   
20767            420      2007             0    98008  47.5724 -122.104   
21201            930      2014             0    98056  47.5285 -122.205   
21576           1310      2007             0    98008  47.5943 -122.110   

       sqft_living15  sqft_lot15  living_sqrt_percent  
49              2680       72513             0.042351  
246             3120       18841             0.436916  
300             4550       19508             0.244086  
457             2280       75396             0.093908  
656             2730       26324             0.070346  
...              ...         ...                  ...  
18914           2980       18782             0.160266  
20325           3550        5634             0.410004  
20767           2670        8525             0.533333  
21201           3560        5762             0.667377  
21576           3470       18270             0.458239  

[76 rows x 22 columns]

У клиента 76 вариантов

### 5.2 Выберите дом клиенту
Клиент хочет дом либо с очень красивым видом из окна, либо с видом на набережную, в очень хорошем состоянии и год постройки не меньше 1980 года. В какой ценовом диапазоне будут дома?

In [204]:
df_max = df[
    ((df["waterfront"] == 1) | (df["view"] == 4)) &
    (df["condition"] > 4.5) &
    (df["yr_built"] > 1978)
]['price'].max()

df_min = df[
    ((df["waterfront"] == 1) | (df["view"] == 4)) &
    (df["condition"] > 4.5) &
    (df["yr_built"] > 1978)
]['price'].min()

print(df_min, df_max)

1295000.0 3000000.0


Цена домов будет от 1295000 до 3000000

### 5.3 Выберите дом клиенту
Клиент хочет дом без подвала, с двумя этажами, стоимостью до 150000. Какая оценка по состоянию у таких домов в среднем?

In [205]:
condition = df[
    (df["sqft_basement"] == 0) &
    (df["floors"] ==2) &
    (df["price"] < 150000)
]['condition'].mean()
condition

2.8333333333333335

Такие дома имеют оценку 2.83 по состоянию